### PR TITLE
Extend listener auth handler APIs for `http:Headers` class

### PR DESCRIPTION
**Pull request by ldclakmal**
## Purpose
This PR extends the listener auth handler APIs for `http:Headers` class.

Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/1013

Related to: https://github.com/ballerina-platform/ballerina-standard-library/issues/584

### CODE REVIEW

**Review by ayeshLK — APPROVED** [review]

File: http-ballerina/auth_listener_oauth2_handler.bal [hunk 1]
```diff
@@ -40,11 +40,11 @@ public client class ListenerOAuth2Handler {
 
     # Authorizes with the relevant authentication & authorization requirements.
     #
-    # + data - The `http:Request` instance or `string` Authorization header
+    # + data - The `http:Request` instanceor `http:Headers` instance  or `string` Authorization header
```
Comment (chamil321) [inline http-ballerina/auth_listener_oauth2_handler.bal:43]:
```suggestion
    # + data - The `http:Request` instance or `http:Headers` instance or `string` Authorization header
```

File: http-ballerina/auth_utils.bal [hunk 2]
```diff
@@ -87,17 +87,24 @@ isolated function buildCompleteErrorMessage(error err) returns string {
...
+    } else if (data is Headers) {
         string|HeaderNotFoundError header = data.getHeader(AUTH_HEADER);
         if (header is string) {
             return regex:split(header, " ")[1];
         } else {
             return prepareListenerAuthError("Authorization header not available.", header);
         }
+    } else {
+        return regex:split(<string>data, " ")[1];
     }
```
Comment (chamil321) [inline http-ballerina/auth_utils.bal:108]: Shall we try this?
```ballerina
if (data is string) {
    return regex:split(<string>data, " ")[1];
} else {
    object {
        public function getHeader() returns string|HeaderNotFoundError;
    } headers = data;
    var header = headers.getHeader(AUTH_HEADER);
    if (header is string) {
        return regex:split(header, " ")[1];
    } else {
        return prepareListenerAuthError("Authorization header not available.", header);
    }
}
```

Comment (ldclakmal) [reply http-ballerina/auth_utils.bal]: Fixed with https://github.com/ballerina-platform/module-ballerina-http/pull/247/commits/2d03298cdc8a903a619faa03cc739072c4e5b8cb

File: http-ballerina/auth_utils.bal [hunk 3]
```diff
@@ -87,12 +87,15 @@ isolated function buildCompleteErrorMessage(error err) returns string {
...
-// Extract the credential from `http:Request` or `string` header.
-isolated function extractCredential(Request|string data) returns string|ListenerAuthError {
+// Extract the credential from `http:Request`, `http:Headers` or `string` header.
+isolated function extractCredential(Request|Headers|string data) returns string|ListenerAuthError {
     if (data is string) {
         return regex:split(<string>data, " ")[1];
     } else {
-        string|HeaderNotFoundError header = data.getHeader(AUTH_HEADER);
+        object {
+            public function getHeader() returns string|HeaderNotFoundError;
```
Comment (chamil321) [inline http-ballerina/auth_utils.bal:96]:
```suggestion
            public function getHeader(string headerName) returns string|HeaderNotFoundError;
```

File: http-ballerina/auth_utils.bal [hunk 4]
```diff
@@ -87,12 +87,15 @@ isolated function buildCompleteErrorMessage(error err) returns string {
...
-// Extract the credential from `http:Request` or `string` header.
-isolated function extractCredential(Request|string data) returns string|ListenerAuthError {
+// Extract the credential from `http:Request`, `http:Headers` or `string` header.
+isolated function extractCredential(Request|Headers|string data) returns string|ListenerAuthError {
     if (data is string) {
         return regex:split(<string>data, " ")[1];
     } else {
-        string|HeaderNotFoundError header = data.getHeader(AUTH_HEADER);
+        object {
+            public function getHeader(string headerName) returns string|HeaderNotFoundError;
```
Comment (chamil321) [inline http-ballerina/auth_utils.bal:96]:
```suggestion
            public isolated function getHeader(string headerName) returns string|HeaderNotFoundError;
```

**Review by chamil321 — APPROVED** [review]

**Pull request merged**